src/odemis/driver/orsay.py

```diff
@@ -2656,22 +2687,255 @@ def __init__(self, name, role, parent, **kwargs):
         • power: IntEnumerated, choices={0: "off", 1: "on"}
         • blanker: VAEnumerated, choices={True: "blanking", False: "no blanking", None: "imaging"}
         """
-
         super().__init__(name, role, parent=parent, **kwargs)
+        self._shape = (1024, 1024)
 
+        if not hasattr(self.parent, "_fib_beam"):
+            raise ValueError(f"To create a Orsay scanner component the parent should also have the FIBBeam child.")
         self._fib_beam = self.parent._fib_beam  # reference to the FIBBeam object
+
+        if not hasattr(self.parent, "_fib_source"):
+            raise ValueError(f"To create a Orsay scanner component the parent should also have the FIBSource child.")
         self._fib_source = self.parent._fib_source  # reference to the FIBSource object
-        # In case there is no such sibling, let it raise an exception
 
+        # In case there is no such sibling, let it raise an exception
         self.blanker = self._fib_beam.blanker
         self.power = self._fib_source.gunOn
 
+        self.horizontalFOV = self._fib_beam.horizontalFOV
+        self.scale = model.VAEnumerated((self.shape[0] / self._fib_beam.imageFormat.value[0],
+                                         self.shape[1] / self._fib_beam.imageFormat.value[1]),
+                                        unit="", choices={(1.0, 1.0), (2.0, 2.0)},
+                                        setter=self._setScale, getter=self._getScale)
+        self.resolution = self._fib_beam.resolution
+        self.translation = self._fib_beam.translation
+
+        self.rotation = self._fib_beam.rotation
+        self.accelVoltage = self._fib_source.acceleratorVoltage
+
+        self.dwellTime = self._fib_beam.dwellTime
+        self.pixelSize = model.TupleContinuous((self._fib_beam.horizontalFOV.value / self.shape[0],
+                                                self._fib_beam.horizontalFOV.value / self.shape[1]),
+                                               unit="m", range=((0.0, 0.0), (1 / 1024, 1 / 1024)))
+
+        # Find all available presets:
+        self.presetData = self.getAllPresetData()
+
+        # Create a preset mask with all the relevant components for the presets. (And maybe some extra)
+        relevant_components = (self.parent.datamodel.HybridAperture,
+                               self.parent.datamodel.HVPSFloatingIon,
+                               self.parent.datamodel.IonColumnMCS,
+                               self.parent.datamodel.Sed,
+                               self.parent.datamodel.HybridPlatform,
+                               self.parent.datamodel.Miss,)
+        # In case the preset mask already exist it is automatically overwritten.
+        self.parent.preset_manager.CreatePresetMask("Odemis-preset-mask", *relevant_components)
+        self.completePresetMaskName = "Odemis-preset-mask"  # Preset defined to load all the relevant components.
```

Comment:
wait on what to do in response on orsay

Reply:
How do we define this outside the class? This needs the componenets/parameters as input which passed. We could make the name a constant outside the class but since it is created in this class only it makes more sense to make it a constant at least somewhat related to this class. Or is that what you mean?

Reply:
Yes, I just mean the name should be a constant. Define it either just before the class or at the top of the file.